client/src/pages/Demo/Demo.js

```diff
@@ -1,10 +1,43 @@
-import React from 'react';
-import { Container } from 'react-bootstrap';
+import React, {useReducer, useState, useEffect} from 'react';
+import { Container, Button } from 'react-bootstrap';
 import { ProgressCircle } from '../../components/progress-circle/ProgressCircle';
 import { Color } from '../../helpers/enums/Colors';
 import { ProductTable } from '../../components/product-table/ProductTable';
+import {GenericModal} from "../../components/Modal";
+import {ProductAdd} from "../../components/product-add/ProductAdd";
+
+const initialModalState = {isModalOpen: false};
+
+function reducer(state, action) {
+    switch (action.type) {
+        case 'OPEN_MODAL':
+            return {isModalOpen: true};
+        case 'CLOSE_MODAL':
+            return {isModalOpen: false};
+        default:
+            throw new Error();
+    }
+}
+
+export const useOpenModal = () => {
+
+    const [state, dispatch] = useReducer(reducer, initialModalState);
```

Comment:
Ogólnie ten stan powinien być podpięty pod kontekst i tam powinno zostać dodane nowe pole isModalOpen: false. W tym reducerze co już jest należy to wykorzystać (./client/src/context/**). Należy tak zrobić ponieważ własne hooki mają stan pod dany komponent gdzie są używane.  W typ wypadku zamist useReducer powinno być const {dispatch} =  useContext (AuthorizationContext). Nazwę AuthorizationContext możesz zmienić na GlobalContext.  A w pliku /client/src/context/index.js dodać ten useReducer, który będzie działać na stanie GlobalContext i na funkcji dipsatch która już istnieje, należy tam jedynie dodać nowe akcje OPEN_MODAL i CLOSE ... Należy również pamiętać żeby zwracać poprzedni stan plus ten nowy {... state, isModalOpen: true}.